packages/core/src/modules/credentials/repository/CredentialExchangeRecord.ts

```diff
@@ -45,6 +48,7 @@ export class CredentialExchangeRecord extends BaseRecord<DefaultCredentialTags,
   public connectionId?: string
   public threadId!: string
   public state!: CredentialState
+  public role!: CredentialRole
```

Comment:
I'm not really a fan of having tags without storing it in the record. If you then want to access it, you have to retrieve it from the tags. 

Tags are only for querying records, and this is broader. E.g. if you want to show issued vs received credentials, both will have state done. Knowing the role is important in that case.

Reply:
And not metadata because it's part of the core protocol to have a role and we have it for other protocols in place already as non-metadata. So this feels the most consistent to me

Reply:
Okay that makes sense to me! Thanks for the explanation.